variables.tf

```diff
@@ -41,67 +47,14 @@ variable "prevent_destroy" {
   default     = true
 }
 
-variable "purpose" {
-  type        = string
-  description = "The immutable purpose of the CryptoKey. Possible values are ENCRYPT_DECRYPT, ASYMMETRIC_SIGN, and ASYMMETRIC_DECRYPT."
-  default     = "ENCRYPT_DECRYPT"
-}
-
-variable "set_owners_for" {
-  description = "Name of keys for which owners will be set."
-  type        = list(string)
+variable "acl" {
+  description = "Access control list for the managed keys."
+  type        = list(any)
```

Comment:
Can we enrich this (and key_opts) with some info as comments about the expected object similar to https://github.com/terraform-google-modules/terraform-google-cloud-storage/blob/d217c7535a6745dcccd1dbf87524b11786627360/modules/simple_bucket/variables.tf#L98-L100

Reply:
Yup, we could, however acl `[]` list allows to add multiple owners to the same key if given/met multiple times. And I would actually prefer to keep it decoupled, i.e. key options is key options while iam permissions binding is a bit different story.

Reply:
@dennybaa I meant we should just add a comment  within the type block which will allow tfdocs to surface this in the [readme](https://github.com/terraform-google-modules/terraform-google-cloud-storage/tree/d217c7535a6745dcccd1dbf87524b11786627360/modules/simple_bucket#inputs:~:text=no-,lifecycle_rules,-The%20bucket%27s%20Lifecycle).

```
list(
# objects with keys
# key = key name
# owners = optional list of owners
# ....
any)
```